src/cpp/process_image.cpp

```diff
@@ -0,0 +1,102 @@
+// Process blossom image
+
+#include <filesystem>
+#include <string>
+
+#ifndef cv
+#include <opencv2/opencv.hpp>
+#endif
+
+#include "process_image.hpp"
+
+
+// Defines
+#define IMAGEPATH "images/image"
+#define EXTENSION ".jpg"
+
+
+// Function prototypes
+std::string getOsName();
+
+
+// Here is where the image gets processed
+std::string ProcessImage::get_processed_image()
+{
+    // Variables
+    static int count = 0;
+    count++;
+    std::string originalImagePath = IMAGEPATH + std::to_string(count) + EXTENSION;
+    std::string processedImagePath = "images/processed_image" + std::to_string(count) + EXTENSION;
+    std::string cachePath = std::filesystem::temp_directory_path().string();
+
+    if (getOsName() == "Apple")
+    {
+        if (cachePath.at(cachePath.length() - 1) == '/')
+        {
+            // cachePath += '/';// !=
+            cachePath = cachePath.substr(0, cachePath.find_last_of("/"));// ==
+        }
+    
+        // Edit path to be cache directory
+        cachePath = cachePath.substr(0, cachePath.find_last_of("/"));
+        cachePath += "/Library/Caches/";
+        // return "error: Good";
+    }
+    else
+    {
+        return "error: Android not yet supported";
```

Comment:
I still need to set up the Android C++ link before openCV will work with Android.